test_wazuh/test_fim/test_ambiguous_confs/test_ambiguous_simple.py

```diff
@@ -215,20 +239,58 @@ def test_ambiguous_recursion(dirname, recursion_level, tags_to_apply,
 
     min_timeout = DEFAULT_TIMEOUT
     scheduled = get_configuration['metadata']['fim_mode'] == 'scheduled'
-
+    
     recursion_subdir = 'subdir'
     path = dirname
-    for n in range(recursion_level):
-        path = os.path.join(path, recursion_subdir + str(n + 1))
-        regular_file_cud(path, wazuh_log_monitor,
-                         time_travel=scheduled,
-                         min_timeout=min_timeout, triggers_event=True)
 
-    for n in list(range(recursion_level, 4)):
-        path = os.path.join(path, recursion_subdir + str(n + 1))
-        regular_file_cud(path, wazuh_log_monitor,
-                         time_travel=scheduled,
-                         min_timeout=min_timeout, triggers_event=False)
+    # Iterate from ini to fin and verify that events are generated in the nested directories.
+    path = _test_recursion_cud(ini=0, fin=recursion_level, path=path,
+                        recursion_subdir=recursion_subdir,
+                        scheduled=scheduled,
+                        min_timeout=min_timeout, triggers_event=True)
+
+    # Iterate from ini to fin and verify that events are NOT generated in nested directories
+    # beyond the established recursion level.
+    _test_recursion_cud(ini=recursion_level, fin=4, path=path,
```

Comment:
I would use dynamic lenght for the recursion such as fin=recursion_level +  1 or recursion_level + 2 so the test works with any given config and not only the actual one.